addon/validations/factory.js

```diff
@@ -133,94 +126,55 @@ function processDefaultOptions(validations = {}) {
 }
 
 /**
- * Create the global properties under the validations object.
- * These are computed collections on different properties of each attribute validations CP
- * @method createGlobalValidationProps
+ * Creates the validations object that will become `model.validations`. Creates all necessary global CPs such as
+ * `isValid`, `isAsync`, etc and also the `attrs` object.
+ * @method createValidationsObject
  * @private
- * @param  {Array} validatableAttrs
- * @return {Object}
+ * @param  {Object} validations
+ * @return {Ember.Object}
  */
-function createGlobalValidationProps(validatableAttrs) {
-  var props = {};
-  props.isValid = and(...validatableAttrs.map((attr) => `attrs.${attr}.isValid`)).readOnly();
-  props.isValidating = or(...validatableAttrs.map((attr) => `attrs.${attr}.isValidating`)).readOnly();
-  props.isDirty = or(...validatableAttrs.map((attr) => `attrs.${attr}.isDirty`)).readOnly();
-  props.isAsync = or(...validatableAttrs.map((attr) => `attrs.${attr}.isAsync`)).readOnly();
-  props.isNotValidating = not('isValidating').readOnly();
-  props.isInvalid = not('isValid').readOnly();
-  props.isTruelyValid = and('isValid', 'isNotValidating').readOnly();
-
-  props._promise = computed(...validatableAttrs.map((attr) => `attrs.${attr}._promise`), function() {
-    var promises = [];
-    validatableAttrs.forEach((attr) => {
-      var validation = get(this, `attrs.${attr}`);
-      if (get(validation, 'isAsync')) {
-        promises.push(get(validation, '_promise'));
-      }
-    });
-    return RSVP.all(flatten(promises));
-  });
+function createValidationsObject(validations = {}) {
+  return Ember.Object.extend({
+    model: null,
+    attrs: null,
+    isValidations: true,
 
-  props.messages = computed(...validatableAttrs.map((attr) => `attrs.${attr}.messages`), function() {
-    var messages = [];
-    validatableAttrs.forEach((attr) => {
-      var validation = get(this, `attrs.${attr}`);
-      if (validation) {
-        messages.push(get(validation, 'messages'));
-      }
-    });
+    // Caches
+    _validators: {},
+    _debouncedValidations: {},
 
-    return emberArray(flatten(messages)).compact();
-  });
+    // Private
+    _validationRules: null,
+    _validatableAttributes: null,
 
-  props.message = computed('messages.[]', cycleBreaker(function() {
-    return get(this, 'messages.0');
-  }));
+    validate,
+    validateSync,
 
+    init() {
+      this._super(...arguments);
+      let inheritedValidations = this.get('model').constructor.superclass.prototype.get('validations');
```

Comment:
This line is pretty sketchy. Before it was `this.get('model')._super()` but that doesnt work for Ember >= 2.1.